server/src/user/service.js

```diff
@@ -1,10 +1,16 @@
 const UserModel = require('./model')
 const bcrypt = require('bcryptjs')
 const errorHandler = require('../util/errorUtil')
+const mailer = require('../services/mailer')
+const crypto = require('crypto')
 
 exports.add = (req, res) => {
+  const currentDate = (new Date()).valueOf().toString()
+  const random = Math.random().toString()
+  const token = crypto.createHash('sha1').update(currentDate + random).digest('hex')
   let newUser = new UserModel({
     email: req.body.email,
+    token: token,
```

Comment:
In fact you don't need all variables before like `currentDate`, `random` and `token`, you can do it inline, this makes it not use to String since you use concatenation it wil be casted directly.

Reply:
done